src/analysis/useComputeImages.ts

```diff
@@ -0,0 +1,67 @@
+import { LoadedState } from '@terra-ui-packages/core-utils';
+import { useEffect } from 'react';
+import { RuntimeToolLabel } from 'src/analysis/utils/tool-utils';
+import { ComputeImageProvider } from 'src/libs/ajax/compute-image-providers/ComputeImageProvider';
+import { useLoadedData } from 'src/libs/ajax/loaded-data/useLoadedData';
+import { useCancellation, useStore } from 'src/libs/react-utils';
+import { workspaceStore } from 'src/libs/state';
+import { isGoogleWorkspaceInfo, WorkspaceInfo, WorkspaceWrapper } from 'src/libs/workspace-utils';
+
+export interface ComputeImage {
+  id: string;
+  url: string;
+  isCommunity: boolean;
+  isRStudio: boolean;
+  isTerraSupported: boolean;
+  toolLabel?: RuntimeToolLabel;
+  label: string;
+  packages: string;
+  requiresSpark: boolean;
+  updated: string;
+  version: string;
+}
+
+export interface ComputeImageStore {
+  refresh: () => void;
+  loadedState: LoadedState<ComputeImage[], unknown>;
+}
+
+export const useComputeImages = (): ComputeImageStore => {
+  const signal = useCancellation();
+  const workspace: WorkspaceWrapper = useStore<WorkspaceWrapper>(workspaceStore);
+  const [loadedState, setLoadedState] = useLoadedData<ComputeImage[]>({
+    onError: (state) => {
```

Comment:
Should this (or something) show an error notification to the user if images can't be loaded?

Reply:
Yes, probably, but I'm ok with this for now.